src/extension/kernel.ts

```diff
@@ -1119,4 +1119,79 @@ export class Kernel implements Disposable {
   public getPlainCache(cacheId: string): Promise<Uint8Array> | undefined {
     return this.serializer?.getPlainCache(cacheId)
   }
+
+  async runProgram(program?: RunProgramOptions | string) {
+    let programOptions: RunProgramOptions
+
+    if (!this.runner) {
+      return Promise.reject(new Error('No runner available'))
+    }
+
+    if (typeof program === 'object') {
+      programOptions = program
+    } else if (typeof program === 'string') {
+      programOptions = {
+        programName: '',
+        background: false,
+        exec: {
+          type: 'script',
+          script: program,
+        },
+        languageId: 'sh',
+        storeLastOutput: false,
+        tty: true,
+      }
+    } else {
+      return Promise.reject(new Error('Invalid runProgram arguments'))
+    }
+
+    const programSession = await this.runner.createProgramSession(programOptions)
+    this.context.subscriptions.push(programSession)
+
+    let execRes: string | undefined
+    const onData = (data: string | Uint8Array) => {
+      if (execRes === undefined) {
+        execRes = ''
+      }
+      execRes += data.toString()
+    }
+
+    programSession.onDidWrite(onData)
+    programSession.onDidErr(onData)
+    await programSession.run()
```

Comment:
You can't await this before "wiring up" the promise. It'll block here indefinitely. That's likely why the callee can't currently be awaited. It just hangs.